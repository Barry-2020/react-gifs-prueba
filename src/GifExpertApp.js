import React, { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";
export const GifExpertApp = ({ defaultCategories = []}) => {
    // const categories = ['One Puch', 'Pokemon', 'Dragon Ball'];

    const [categories, setCategories] = useState(defaultCategories);


    // const handleAdd = () =>{
    //     // setCategories( [...categories, 'probando...']);
    //     setCategories( cats => [...cats,'xd'])
    // }

    return(
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={ setCategories }/>
            <hr/>
            <ol>
                {
                    categories.map( (category, i) => (
                        <GifGrid 
                            key = {category}
                            category={category}></GifGrid>
                    ))
                }
            </ol>
        </>
    )
}