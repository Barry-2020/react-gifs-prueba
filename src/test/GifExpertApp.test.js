import { shallow } from "enzyme";
import '@testing-library/jest-dom';
import { GifExpertApp } from "../GifExpertApp";

describe('Pruebas gifExpertApp', () => {


    let wrapper = shallow( <GifExpertApp/>);
    test('Debe de mostrar correctamente', () => {

        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar una lista de categorias', () => {
        const catergories = ['One punch', 'Dragon ball'];
        let wrapper = shallow( <GifExpertApp defaultCategories={ catergories }/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(catergories.length);
    })
})
