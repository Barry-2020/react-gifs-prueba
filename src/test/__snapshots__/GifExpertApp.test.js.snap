// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas gifExpertApp Debe de mostrar correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas gifExpertApp debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One punch"
      key="One punch"
    />
    <GifGrid
      category="Dragon ball"
      key="Dragon ball"
    />
  </ol>
</Fragment>
`;
